assets/components/src/plugin-toggle/index.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * Plugin toggle group component.
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { Component, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import { ActionCard, Waiting } from '../';
+import './style.scss';
+
+/**
+ * Plugin toggle group component.
+ */
+class PluginToggle extends Component {
+	state = {
+		pluginInfo: {},
+	};
+
+	componentDidMount = () => {
+		this.retrievePluginInfo();
+	};
+
+	/**
+	 * Retrieve complete data about Newspack plugins.
+	 */
+	retrievePluginInfo = () => {
+		const { plugins } = this.props;
+		return new Promise( ( resolve, reject ) => {
+			apiFetch( { path: '/newspack/v1/plugins/' } ).then( pluginInfo =>
+				this.setState( { pluginInfo } )
+			);
+		} );
+	};
+
+	/**
+	 * Install/activate or remove a plugin.
+	 */
+	managePlugin = ( plugin, value ) => {
+		const { plugins } = this.props;
+		const { pluginInfo } = this.state;
+		const action = value ? 'configure' : 'uninstall';
+		const params = {
+			path: `/newspack/v1/plugins/${ plugin }/${ action }/`,
+			method: 'post',
+		};
+
+		this.setState(
+			{
+				pluginInfo: { ...pluginInfo, [ plugin ]: { ...pluginInfo[ plugin ], inFlight: action } },
+			},
+			() => {
+				apiFetch( params ).then( response => {
+					const { shouldRefreshAfterUpdate } = plugins[ plugin ];
+					this.setState( { pluginInfo: { ...pluginInfo, [ plugin ]: response } }, () => shouldRefreshAfterUpdate && location.reload() );
+				} );
+			}
+		);
+	};
+
+	/**
+	 * Render.
+	 */
+	render() {
+		const { plugins } = this.props;
+		const { pluginInfo } = this.state;
+		return this.prepareDataForRender( plugins, pluginInfo ).map( plugin => {
+			const { name, description, href, slug, status, editPath, inFlight } = plugin;
+			const pluginStatus = this.statusForPlugin( plugin );
+			const handoff = ! href && pluginStatus && editPath ? slug : null;
+			return (
+				<ActionCard
```

Comment:
The output of this is an array of components, so a `key` prop is mandatory on a component. On first pass the `plugin` here is an empty object - which gives us no unique value - so simply the index of the iteration can be used

Reply:
Good catch. https://github.com/Automattic/newspack-plugin/pull/406/commits/7627c488e9f8c98012c9d6d23bee20f6d6b64cfd